lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs

```diff
@@ -2932,6 +2956,251 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
+            (Link.createUnsignedTransaction @'Shelley wa) Default payload
+        verify rTx
+            [ expectResponseCode HTTP.status202
+            ]
+
+        let txCbor = getFromResponse #transaction rTx
+        let decodePayload = Json (toJSON $ ApiSerialisedTransaction txCbor)
+
+        let policyWithHash = Link.getPolicyKey @'Shelley wa (Just True)
+        (_, policyKeyHashPayload) <-
+                unsafeRequest @ApiPolicyKey ctx policyWithHash Empty
+        let (Just policyKeyHash) =
+                keyHashFromBytes (Payment, getApiPolicyKey policyKeyHashPayload)
+
+        let tokenPolicyId' =
+                UnsafeTokenPolicyId . Hash $
+                unScriptHash $
+                toScriptHash $
+                RequireAllOf [RequireSignatureOf policyKeyHash, ActiveFromSlot 120]
+        let tokens = TokenMap.singleton
+                (AssetId tokenPolicyId' tokenName')
+                (TokenQuantity 50_000)
+        let mintScript =
+                ( ApiT tokenPolicyId'
+                , ApiT (RequireAllOf [RequireSignatureOf policyKeyHash,ActiveFromSlot 120])
+                )
+
+        let activeAssetsInfo = ApiAssetMintedBurned
+                { tokenMap = ApiT tokens
+                , policyScripts = [mintScript]
+                , walletPolicyKeyHash = policyKeyHashPayload
+                }
+        let inactiveAssetsInfo = ApiAssetMintedBurned
+                { tokenMap = ApiT TokenMap.empty
+                , policyScripts = []
+                , walletPolicyKeyHash = policyKeyHashPayload
+                }
+
+        rDecodedTx <- request @(ApiDecodedTransaction n) ctx
+            (Link.decodeTransaction @'Shelley wa) Default decodePayload
+        verify rDecodedTx
+            [ expectResponseCode HTTP.status202
+            , expectField #assetsMinted (`shouldBe` activeAssetsInfo)
+            , expectField #assetsBurned (`shouldBe` inactiveAssetsInfo)
+            ]
+
+    it "TRANS_NEW_CREATE_10e - Burning assets" $ \ctx -> runResourceT $ do
+        wa <- fixtureWallet ctx
+
+        let (Right tokenName') = mkTokenName "ab12"
+
+        let payload = Json [json|{
+                "minted_burned": [{
+                    "policy_script_template":
+                        { "all":
+                           [ "cosigner#0",
+                             { "active_from": 120 }
+                           ]
+                        },
+                    "asset_name": #{toText tokenName'},
+                    "operation":
+                        { "burn" :
+                              { "quantity": 50000,
+                                "unit": "assets"
+                              }
+                        }
+                }]
+            }|]
+
+        rTx <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shelley wa) Default payload
+        verify rTx
+            [ expectResponseCode HTTP.status202
```

Comment:
I think burning should fail in this case with the error message saying that there are no tokens to burn. This transaction would fail anyway on sign/submit since there are no tokens with wallet's `policy_id.asset_name` on wallet's balance.

Reply:
of course it would fail after submission. this is construction phase. I would not outlaw it in this PR as it is not small. But we can think about this later.

Reply:
Just to note, it is similar case as if you try to construct transaction from an empty wallet... you'll get `not_enough_money` error in that case. The same behavior should be applied here I think. If you try to construct burning tx on the wallet that has nothing to burn you should get proper error on construction phase.